pkg/services/ngalert/state/template_test.go

```diff
@@ -0,0 +1,392 @@
+	}, {
+		name:     "humanizePercentage - float64",
+		text:     "{{ -0.22222 | humanizePercentage }}:{{ 0.0 | humanizePercentage }}:{{ 0.1234567 | humanizePercentage }}:{{ 1.23456 | humanizePercentage }}",
+		expected: "-22.22%:0%:12.35%:123.5%",
+	}, {
+		name:     "humanizePercentage - string",
+		text:     `{{ "-0.22222" | humanizePercentage }}:{{ "0.0" | humanizePercentage }}:{{ "0.1234567" | humanizePercentage }}:{{ "1.23456" | humanizePercentage }}`,
+		expected: "-22.22%:0%:12.35%:123.5%",
+	}, {
+		name:          "humanizePercentage - string with error",
+		text:          `{{ "invalid" | humanizePercentage }}`,
+		expectedError: errors.New(`error executing template __alert_test: template: __alert_test:1:91: executing "__alert_test" at <humanizePercentage>: error calling humanizePercentage: strconv.ParseFloat: parsing "invalid": invalid syntax`),
+	}, {
+		name:     "humanizeTimestamp - float64",
+		text:     "{{ 1435065584.128 | humanizeTimestamp }}",
+		expected: "2015-06-23 13:19:44.128 +0000 UTC",
+	}, {
+		name:     "humanizeTimestamp - string",
+		text:     `{{ "1435065584.128" | humanizeTimestamp }}`,
+		expected: "2015-06-23 13:19:44.128 +0000 UTC",
+	}, {
+		name:     "title",
+		text:     `{{ "aa bb CC" | title }}`,
+		expected: "Aa Bb CC",
+	}, {
+		name:     "toUpper",
+		text:     `{{ "aa bb CC" | toUpper }}`,
+		expected: "AA BB CC",
+	}, {
+		name:     "toLower",
+		text:     `{{ "aA bB CC" | toLower }}`,
+		expected: "aa bb cc",
+	}, {
+		name:     "match",
+		text:     `{{ match "a+" "aa" }} {{ match "a+" "b" }}`,
+		expected: "true false",
+	}, {
+		name:     "match",
+		text:     `{{ match "a+" "aa" }} {{ match "a+" "b" }}`,
+		expected: "true false",
+	}, {
+		name:     "regex replacement",
+		text:     "{{ reReplaceAll \"(a)b\" \"x$1\" \"ab\" }}",
+		expected: "xa",
+	}, {
+		name:     "pass multiple arguments to templates",
+		text:     `{{define "x"}}{{.arg0}} {{.arg1}}{{end}}{{template "x" (args 1 "2")}}`,
+		expected: "1 2",
+	}, {
+		name:     "pathPrefix",
+		text:     "{{ pathPrefix }}",
+		expected: pathPrefix,
+	}, {
+		name:     "externalURL",
+		text:     "{{ externalURL }}",
+		expected: externalURL.String(),
+	}, {
+		name:     "check that query, first and value don't error or panic",
+		text:     "{{ query \"1.5\" | first | value }}",
+		expected: "",
+	}, {
+		name:     "check that label doesn't error or panic",
+		text:     "{{ query \"metric{instance='a'}\" | first | label \"instance\" }}",
+		expected: "",
+	}, {
+		name:     "check that graphLink returns an empty string",
+		text:     "{{ graphLink \"up\" }}",
+		expected: "",
+	}, {
+		name:     "check that tableLink returns an empty string",
+		text:     "{{ tableLink \"up\" }}",
+		expected: "",
+	}, {
+		name:     "check that sortByLabel or strvalue don't error or panic",
```

Comment:
Ditto for `strvalue`, no documentation nor part of #39662

Reply:
Can we add a test case for this too? I have a feeling we should support this label if don't (I'm pretty sure we do) cc: @gerobinson 

```
			"strvalue": func(s *sample) string {
				return s.Labels["__value__"]
			},
```